Kursach

Lelbry opened the pull request:
Тема курсовой это футбольная команда. У меня ж будет турнир между 4 футбольными командами на выбывание, то есть по итогу будет сыграно 3 матча (2024 02 28, 2024 03 01, 2024 03 08 (Финал) соответственно).  
 В приложенном файле (внизу) начал заполнять данные для таблицы Matches* - куда я записываю только значения игроков 1 раз за весь матч (подразумевается в случае замены игрока либо ж если кончился сам матч). Если игроки делали хотя б одно из результативных действий (Голы, Ассисты, Одна из двух карточек желтая или красная или фолы). 
 Изменения 06.04.2024:
- Добавлен триггер на запрет удаления таблиц
- Каждая таблица теперь создается в отдельном скрипте
- Каждые изменения таблиц прописаны в отдельных скриптах
- Обновлена функция Функция для подсчета общей статистики
- Обновлена Процедура обновления возраста игрока, где дополнительно существует проверка на наличе игрока в таблице

Commit:
Курсовая добавлены скрипты для создания таблиц

Commit:
Созданы дополнительно загатовки под будущие задачи курсовой

Lelbry committed:
Переработана структура Курсовой, начато заполнение таблицы Matches

Lelbry committed:
Исправлены опечатки, полностью заполнена таблица Matches

Lelbry committed:
Созданы функции для курсовой работы.

- Список всех игроков из команды (Имя команды)
- Общая статистика по команде (id)
- Подсчёт среднего возраста в команде (id)

Lelbry committed:
Добавлена процедура к курсовой работе и пустой файл для будущего триггера.

MarinaPaley commented on Курсовая работа/01_Создание таблиц Teams, Players, Matches.sql:
```diff
@@ -0,0 +1,26 @@
+--DROP TABLE "Teams";
```
Каждая таблица должна создаваться в отдельном скрипте.

MarinaPaley commented on Курсовая работа/02_Ограничения для таблиц Teams, Players, Matches.sql:
```diff
@@ -0,0 +1,20 @@
+--Ограничения в таблицах
```
Для изменения каждой таблицы также должны быть созданы отдельный файлы. (Для каждой таблицы по одному файлу на изменение).

MarinaPaley commented on Курсовая работа/06_Функция для подсчета общей статистики команды.sql:
```diff
@@ -0,0 +1,36 @@
+--Функция для подсчета общего количества голов, голевых передач, желтых и красных карточек в матче по ID команды
+CREATE OR REPLACE FUNCTION TotalStatsInMatch("team_id" INT)
```
Супер, но я бы, команду передавала по имени.

MarinaPaley commented on Курсовая работа/09_Процедура обновления возраста игрока .sql:
```diff
@@ -0,0 +1,21 @@
+CREATE OR REPLACE PROCEDURE UpdatePlayerAgeByName(player_name VARCHAR, new_age INT)
+AS $$
+BEGIN
+    IF new_age < 15 THEN
```
Я бы спросила, есть дли такой игрок?

MarinaPaley reviewed (changes requested):
Продолжайте...
Триггер. [презентация](https://github.com/MarinaPaley/IOSU-2023/blob/feature/11-1/Programming/%D0%A2%D1%80%D0%B8%D0%B3%D0%B3%D0%B5%D1%80%D1%8B%20(1).pptx)

Lelbry committed:
-Добавлен триггер на запрет удаления таблиц

- Каждая таблица теперь создается в отдельном скрипте
- Каждые изменения таблиц прописаны в отдельных скриптах
- Обновлена функция Функция для подсчета общей статистики
- Обновлена Процедура обновления возраста игрока, где дополнительно существует проверка на наличе игрока в таблице

Lelbry committed:
Добавлены отдельные скрипты для тестировния функционала и другое.

MarinaPaley commented on Курсовая работа/04_Создание представлений view/Создание представления для таблицы Matches.sql:
```diff
@@ -0,0 +1,2 @@
+CREATE VIEW "vMatches" AS
```
Хорошо, но, обычно представления делают для соединения таблиц... Добавьте, пожалуйста.

MarinaPaley reviewed (changes requested):
Все хорошо и правильно.
Добавьте представление для `INNER JOIN` и процедуру, в которой добавляете данные в таблицу Статистика (наверное)

Lelbry committed:
Внесены изменениия:

- Добавлена функция для подсчёта общей статистики отдельного игрока
- Обновлена функция для подсчёта общей статистики команды
- Добавлено тестирование Процедура добавления данных в таблицу Matches
- Добавлена Процедура добавления данных в таблицу Matches
- Обновлено представление view для БД.
